core/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala

```diff
@@ -111,6 +112,24 @@ private[spark] class TypedConfigBuilder[T](
     }
   }
 
+  /** Checks if the user-provided value for the config matches the validator.
+   * If it doesn't match, raise Spark's exception with the given error class. */
+  def checkValue(
+      validator: T => Boolean,
+      errorClass: String,
+      parameters: Map[String, String]): TypedConfigBuilder[T] = {
```

Comment:
do we need this parameter? Looking at the signature of `validator`, `T => Boolean`, I don't think a config definition can provide extra parameters.

Reply:
Sorry, I didn't get how `parameters` are related to the signature of `validator`. Just in case, here is one of examples where we would need the `parameters`:
https://github.com/apache/spark/blob/174a19c1c0391b1084e6a0b8fdbc5c019a662b01/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1128-L1131

And if you remove `parameters` how would you overload the existing one? 
```scala
  def checkValue(validator: T => Boolean, errorMsg: String): TypedConfigBuilder[T] = {
```

We could "inline" the parameters to error message format, but need to migrate all configs in one shot to:
```scala
  def checkValue(validator: T => Boolean, errorClass: String): TypedConfigBuilder[T] = {
```